plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/DeltaLakePageSourceProvider.java

```diff
@@ -320,7 +320,7 @@ else if (column.baseColumnName().equals(ROW_ID_COLUMN_NAME)) {
                 delegateIndex++;
             }
             else if (missingColumnNames.contains(column.baseColumnName())) {
-                transform.constantValue(column.baseType().createNullBlock());
+                transform.constantValue(column.type().createNullBlock());
```

Comment:
How did you find this @raunaqmorarka ?

Reply:
just from debugging the failing product test
it fixes testColumnMappingModeIdAddColumn, but testColumnMappingModeNameAddColumn still fails